compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -142,6 +142,30 @@ def EncodingAttr :
   let genVerifyDecl = 1;
 }
 
+//===---------------------------------------------------------------------===//
+// encoding.matmul_k
+//===---------------------------------------------------------------------===//
+
+def MatmulKAttr : IREEEncoding_Attr<"MatmulK"> {
+  let mnemonic = "matmul_k";
+  let summary = [{An attribute that tracks reduction dimensions for matmul}];
+  let description = [{
+    The attribute is specialized for tracking the matmul reduction dimensions
+    only. The encoded dimensions are the reduction dimensions that will be
+    further used in a matmul.
+
+    Any encoding propagation that transforms reduction dimensions could result
+    in undefined behavior. Because it does not encode the transformations. The
+    information is missing in this context.
```

Comment:
nit: I think this should be either '... in undefined behavior, because' or '... encode the transformations, the information'